### ISSUE TITLE
external, cmake: pydrake.all import broken due to missing fmtlib symbol?

### ISSUE THREAD

**Issue by EricCousineau-TRI** [RobotLocomotion/drake#17744]
### What happened?

see Drake Slack: https://drakedevelopers.slack.com/archives/C270MN28G/p1660831321007099

https://drake-jenkins.csail.mit.edu/view/Production/job/RobotLocomotion/job/drake-external-examples/job/main/507/console
https://drake-jenkins.csail.mit.edu/view/Production/job/RobotLocomotion/job/drake-external-examples/job/main/508/console

\cc @mwoehlke-kitware 

### Relevant log output

```shell
File "{runfiles}/drake/bindings/pydrake/stubgen.py", line 10, in <module>
    import pydrake.all
  File "{runfiles}/drake/bindings/pydrake/all.py", line 31, in <module>
    from .autodiffutils import *
ImportError: {runfiles}/drake/bindings/pydrake/../../_solib_k8/_U_S_Stools_Sinstall_Slibdrake_Cdrake_Ushared_Ulibrary___Utools_Sinstall_Slibdrake/libdrake.so: undefined symbol: 
_ZN6spdlog17pattern_formatterC1ENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEENS_17pattern_time_typeES6_St13unordered_mapIcSt10unique_ptrINS_21custom_flag_formatterESt14default_deleteISA_EESt4hashIcESt8equal_toIcESaISt4pairIKcSD_EEE

...

$ echo _ZN6spdlog17pattern_formatterC1ENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEENS_17pattern_time_typeES6_St13unordered_mapIcSt10unique_ptrINS_21custom_flag_formatterE
St14default_deleteISA_EESt4hashIcESt8equal_toIcESaISt4pairIKcSD_EEE | c++filt
spdlog::pattern_formatter::pattern_formatter(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, spdlog::pattern_time_type, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::unordered_map<char, std::unique_ptr<spdlog::custom_flag_formatter, std::default_delete<spdlog::custom_flag_formatter> >, std::hash<char>, std::equal_to<char>, std::allocator<std::pair<char const, std::unique_ptr<spdlog::custom_flag_formatter, std::default_delete<spdlog::custom_flag_formatter> > > > >)
```

**Comment by jwnimmer-tri** [comment]
(This is probably just the fmt ABI and/or python 3.10 ABI propagation being not yet completed through the nightly builds.)

**Comment by jwnimmer-tri** [comment]
Fixed via #17746.  The original feature request #16987 is still open to track the unrevert.
